lib/kimono_labs_client/web_client.rb

```diff
@@ -0,0 +1,31 @@
+module KimonoLabsClient
+  # Web client for fetching a json endpoint's content via http.
+  class WebClient
+    attr_reader :api_key
+
+    # initializer
+    def initialize(api_key)
+      fail 'API key must be provided!' if api_key.nil?
+
+      @api_key = api_key
+
+      @http = Faraday.new(url: 'https://www.kimonolabs.com') do |http|
+        http.request :url_encoded             # form-encode POST params
+        http.response :logger                  # log requests to STDOUT
+        http.adapter Faraday.default_adapter  # make requests with Net::HTTP
+      end
+    end
+
+    # Lists available web endpoints for a kimono api key.
+    # url format: https://www.kimonolabs.com/kimonoapis?apikey={api_key}
+    # docs: https://www.kimonolabs.com/apidocs#ListApis
+    def list
+      response = @http.get "/kimonoapis?apikey=#{api_key}" # FIXME: use actual URI instance and not a string
```

Comment:
Try [this](http://apidock.com/rails/ActiveSupport/CoreExtensions/Hash/to_query) for constructing the query string portion.